central/debug/service/service.go

```diff
@@ -580,9 +606,21 @@ func (s *serviceImpl) writeZippedDebugDump(ctx context.Context, w http.ResponseW
 			log.Error(err)
 		}
 	}
+}
 
-	if err := zipWriter.Close(); err != nil {
-		log.Error(err)
+func writeTimeoutWarning(zipWriter *zip.Writer) {
+	log.Info("Too much time elapsed in diagnostic bundle. Quitting early as to not to hit timeout.")
+
+	w, err := zipWriter.Create(warningFilename)
+	if err != nil {
+		log.Error(errors.Wrapf(err, "unable to create zip file %q", warningFilename))
+	}
+	warningMessage := "Timeout is reached while creating diagnostic bundle. To increase timeout:\n" +
+		"1) If you use `roxctl`, use `--timeout` flag\n" +
+		"2) If you use API, use `timeout` query parameter: `/api/extensions/diagnostics?timeout=60`"
+	_, err = w.Write([]byte(warningMessage))
```

Comment:
nit: extract this as a package variable and use multi-line strings to get rid of the \n.